src/main/java/esi/backend/controller/RentalController.java

```diff
@@ -24,28 +25,30 @@ public List<Rental> getAllRentals(@PathVariable UUID carId) {
         return rentalService.getAllRentals(carId);
     }
 
-    @RequestMapping("/cars/rentals/{id}")
-    public Optional<Rental> getRental(@PathVariable UUID id) {
-        return rentalService.getRental(id);
+    @RequestMapping("/cars/rentals/{rentalId}")
```

Comment:
The mapping should have `carId` as we discussed in the state model: `Get /cars/{carId}/rentals/{rentalId}` . Same with `deleteRental`.